drivers/usb/device/usb_dc_kinetis.c

```diff
@@ -851,6 +850,15 @@ int usb_dc_ep_mps(const uint8_t ep)
 
 static inline void reenable_all_endpoints(void)
 {
+	struct usb_dc_ep_cfg_data ep_cfg;
+
+	/* Reconfigure control endpoint 0 after a reset */
+	ep_cfg.ep_addr = USB_CONTROL_EP_OUT;
+	ep_cfg.ep_mps = USB_CONTROL_EP_MPS;
+	ep_cfg.ep_type = USB_DC_EP_CONTROL;
+	usb_dc_ep_configure(&ep_cfg);
+	ep_cfg.ep_addr = USB_CONTROL_EP_IN;
+	usb_dc_ep_configure(&ep_cfg);
```

Comment:
I do not remember the reason for the loop below, but you are right, all endpoints are reset, so actually it should be enough to call usb_dc_ep_configure() + usb_dc_enable() for USB_CONTROL_EP_OUT and USB_CONTROL_EP_IN. And rename reenable_all_endpoints(void) to reenable_control_endpoint(void).